### DISCUSSION TITLE
How do you insert components on a scene?

### DISCUSSION THREAD

**Discussion by lukasfri** [bevyengine/bevy#3015] [Q&A] [answered]
How do you apply components such as transform to a scene? There doesn't seem to be a way to do it directly and spawning it as a child like the following example:
```rust
let scene_handle = ass.load::<Scene, &str>("models/tile.glb#Scene0");

commands.spawn_scene(scene_handle.clone());

let mut entity = commands.spawn();
entity
    .borrow_mut()
    .insert(Transform::from_xyz(0.0, -1.0, 0.0))
    .insert(Spawning);

scene_spawner.spawn_as_child(scene_handle, entity.id());
```
doesn't work. It doesn't seem like to be parented when it comes to movement etc. What is the correct way to move around a scene and/or the meshes inside it?

**Comment by wilk10** [comment, marked as answer]
See how it's done in the [update_gltf_scene](https://github.com/bevyengine/bevy/blob/main/examples/3d/update_gltf_scene.rs) example.

It's a bit convoluted indeed, but basically you need to also mark the entities from the scene, not only the parent, and then you can change the components of those entities.
